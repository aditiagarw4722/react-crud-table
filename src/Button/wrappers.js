import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-button');

export const Btn = block('button');

export default { Btn };
