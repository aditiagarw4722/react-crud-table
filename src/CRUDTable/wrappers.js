import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table');

export const Table = block('table');

Table.displayName = 'Table';

Table.Row = element('tr', 'row');
Table.Row.displayName = 'Row';

Table.HeaderCell = element('th', 'header-cell');
Table.HeaderCell.displayName = 'HeaderCell';

Table.Header = element('thead', 'header');
Table.Header.displayName = 'Header';

Table.Body = element('tbody', 'body');
Table.Body.displayName = 'Body';

Table.Cell = element('td', 'cell');
Table.Cell.displayName = 'Cell';

Table.CellLabel = element('div', 'cell-label');
Table.CellLabel.displayName = 'CellLabel';

Table.Caption = element('div', 'caption');
Table.Caption.displayName = 'Caption';

export default { Table };
