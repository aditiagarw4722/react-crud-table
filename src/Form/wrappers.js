import { Form as FormikForm } from 'formik';
import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-form');

export const Form = block(FormikForm);

Form.displayName = 'FormikForm';

Form.Label = element('label', 'label');
Form.Label.displayName = 'Label';

Form.Message = element('div', 'message');
Form.Message.displayName = 'Message';

Form.ErrorMessage = element('div', 'error-message');
Form.ErrorMessage.displayName = 'ErrorMessage';

Form.FieldError = element('div', 'field-error');
Form.FieldError.displayName = 'FieldError';

Form.FieldContainer = element('div', 'field-container');
Form.FieldContainer.displayName = 'FieldContainer';

export default {
  Form,
};
