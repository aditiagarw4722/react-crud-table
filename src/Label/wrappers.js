import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-label');

const Label = block('div');
Label.displayName = 'Label';

export default Label;
