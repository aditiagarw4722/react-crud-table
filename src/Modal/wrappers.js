import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-wrapper');

export const Container = block('div');
Container.BG = element('div', 'background');
Container.Modal = element('div', 'modal');
Container.Title = element('h3', 'title');

export default {
  Container,
};
