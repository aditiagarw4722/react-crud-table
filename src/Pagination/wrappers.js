import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-pagination');

export const Pagination = block('div');

Pagination.Link = element('button', 'link');
Pagination.Prev = element('button', 'previous');
Pagination.Next = element('button', 'next');

export default { Pagination };
