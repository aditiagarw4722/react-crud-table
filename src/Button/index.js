import React from 'react';
import { Btn } from './wrappers';

const Button = (props) => <Btn {...props} />;

export default Button;
