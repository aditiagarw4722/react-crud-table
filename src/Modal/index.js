import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './wrappers';
import Button from '../Button';
import { NO_OP } from '../helpers';

const getDisplay = (visible) => (visible ? 'block' : 'none');

const Modal = (props) => {
  const { children, trigger, title, visible, onHide, onShow } = props;
  const style = {
    display: getDisplay(visible),
  };
  return (
    <div>
      {trigger && (
        <Button modifiers="positive" onClick={onShow}>
          {trigger}
        </Button>
      )}
      <Container style={style}>
        <Container.BG onClick={onHide} />
        <Container.Modal>
          {title && <Container.Title>{title}</Container.Title>}
          {children}
        </Container.Modal>
      </Container>
    </div>
  );
};

Modal.propTypes = {
  title: PropTypes.string,
  onHide: PropTypes.func,
  onShow: PropTypes.func,
  children: PropTypes.node,
  trigger: PropTypes.node,
  visible: PropTypes.bool,
};

Modal.defaultProps = {
  onHide: NO_OP,
  onShow: NO_OP,
  title: '',
  children: null,
  trigger: null,
  visible: false,
};

export default Modal;
