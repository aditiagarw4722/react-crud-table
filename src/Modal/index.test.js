import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import { Container } from './wrappers';
import Button from '../Button';
import Modal from './';

const renderer = new ShallowRenderer();

describe('Modal', () => {
  const props = {
    trigger: 'Open',
    title: 'Test title',
    onShow: jest.fn(),
    onHide: jest.fn(),
    visible: true,
  };

  it('renders as expected with the given props', () => {
    const result = renderer.render(
      <Modal {...props} visible={false}>
        Test
      </Modal>
    );
    expect(result).toMatchSnapshot();
  });

  it('notifies when opened', () => {
    const result = create(<Modal {...props} />);

    const buttons = result.root.findAllByType(Button);

    buttons[0].props.onClick();

    expect(props.onShow).toHaveBeenCalledTimes(1);
  });

  it('notifies when closed', () => {
    const result = create(<Modal {...props} />);

    result.root.findByType(Container.BG).props.onClick();

    expect(props.onHide).toHaveBeenCalledTimes(1);
  });
});
