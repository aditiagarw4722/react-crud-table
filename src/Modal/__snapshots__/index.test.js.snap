// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders as expected with the given props 1`] = `
<div>
  <Button
    modifiers="positive"
    onClick={[MockFunction]}
  >
    Open
  </Button>
  <Wrapper
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Background
      onClick={[MockFunction]}
    />
    <Modal>
      <Button
        modifiers="circular,close"
        onClick={[MockFunction]}
      >
        X
      </Button>
      <Title>
        Test title
      </Title>
      Test
    </Modal>
  </Wrapper>
</div>
`;
